pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -70,9 +65,16 @@ pub struct DiskUsageEvictionTaskConfig {
     pub period: Duration,
 }
 
+#[derive(Default)]
+pub struct State {
+    /// Exclude http requests and background task from running at the same time.
+    mutex: tokio::sync::Mutex<()>,
```

Comment:
Still the same question, why do we need async mutex here if we use only try_lock? 

https://github.com/neondatabase/neon/pull/3809#discussion_r1149364530

> Do we really need async mutex here? The only usage is synchronous try_lock.
> 
> Additionally we can use AtomicBool with defer macro (see an example from proxy here https://github.com/zenithdb/zenith/blob/93f3f4ab5fb589296ea44929f5c113f680777ee6/pageserver/src/page_service.rs#L240)

Reply:
Will be resolved in parent PR I guess

Reply:
Sure